subsys/shell/modules/devmem_service.c

```diff
@@ -26,6 +32,122 @@ static bool littleendian;
 #define CHAR_CAN 0x18
 #define CHAR_DC1 0x11
 
+#ifndef BITS_PER_BYTE
+#define BITS_PER_BYTE 8
+#endif
+
+static int memory_dump(const struct shell *sh, mem_addr_t phys_addr, size_t size, uint8_t width)
+{
+	uint32_t value;
+	size_t data_offset;
+	mm_reg_t addr;
+	const size_t vsize = width / BITS_PER_BYTE;
+	uint8_t data[SHELL_HEXDUMP_BYTES_IN_LINE];
+
+#if defined(CONFIG_MMU) || defined(CONFIG_PCIE)
+	device_map((mm_reg_t *)&addr, phys_addr, size, K_MEM_CACHE_NONE);
```

Comment:
Shouldn't there be a respective `unmap()` call before returning from this function, or it's fine to leave the memory mapped after you're done with it?

Reply:
That's a good question. This part was copied from the existing `devmem read`